src/api/routes/interviews.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * API for interviews.
+ * An interview is always bound to a user.
+ * Therefore, every endpoint requires an authentication.
+ */
+
+module.exports = (db, security) => {
+    const router = require('express').Router()
+    const validate = security.validate()
+    const messageKeys = require('../util/message-keys')
+
+    /**
+     * Get all interviews of a user.
+     * If now interviews are found, an empty array is being returned.
+     */
+    async function getInterviews(req, res) {
+        const userId = req.user.data.id
+        try {
+            const interviews = await db.getInterviews(userId)
+            if (interviews === null || !Array.isArray(interviews)) {
+                return res.status(404).json({
+                    message: 'The passed user id ("' + userId + '") was not found in the database',
+                    key: messageKeys.INVALID_USER_ID
+                })
+            }
+            return res.send(interviews)
+        }
+        catch(e) {
+            return res.status(500).json({
+                message: 'A database error occurred.',
+                key: messageKeys.DB_ERROR
+            })
+        }
+    }
+
+    /**
+     * Get a specific interview of a user by its id.
+     * If the interview with the given id does not exists or does not belong to the user,
+     * status code 404 is sent.
+     */
+    async function getInterview(req, res) {
+        const userId = req.user.data.id
+        const interviewId = req.params.interviewId
+        try {
+            const interview = await db.getInterview(userId, interviewId)
+            if (interview === null || typeof interview !== 'object') {
```

Comment:
As above (`interviews` array check): Make sure the db API is properly specified. A check against `typeof interview !== 'object'` might not be necessary.

Reply:
Check is not necessary and can be removed.